sdk/python/cmd/pulumi-language-python/main.go

```diff
@@ -996,6 +996,7 @@ func (host *pythonLanguageHost) Run(ctx context.Context, req *pulumirpc.RunReque
 		}
 		typecheckerCmd.Stdout = os.Stdout
 		typecheckerCmd.Stderr = os.Stderr
+		typecheckerCmd.Dir = req.Info.ProgramDirectory
```

Comment:
Lang conformance tests were failing here, and previously only passed by accident. Since the tests run from the sdk/python dir, pyrightconfig picked up the virtual environment named "venv" there. When we moved to a virtual environment named ".venv" with the change to uv, they started failing. What we want is for the typechecker to always run in the program directory.